esmvaltool/diag_scripts/aerosols/aero_utils.py

```diff
@@ -0,0 +1,270 @@
+
+    else:
+        for n in np.arange(len(pt_lat)):
+            pt_lat1.append(pt_lat[n])
+            pt_lon1.append(pt_lon[n])
+
+    if len(pt_lat1) != len(pt_lon1):
+        raise AeroAnsError('Extract_pt:Mismatch in number of lat/long values')
+
+    if 'level' in keys and 'height' in keys:
+        raise AeroAnsError('Extract_pt: Both Level and Height requested')
+
+    if icube.coords()[0].name() == 'time':  # only lat,lon,lev accepted
+        raise AeroAnsError(
+            'Extract_pt:Cannot handle time dimension at present')
+
+    if 'level' in keys and not icube.coord(
+            'model_level_number') and not icube.coord('pseudo_level'):
+        raise AeroAnsError('Extract_pt:Level requested, but not found in cube')
+
+    if 'height' in keys:
+        pt_hgt = []
+
+        if not isinstance(keys['height'], list):
+            pt_hgt.append(keys['height'])
+
+        else:
+            for n in np.arange(len(keys['height'])):
+                pt_hgt.append(keys['height'][n])
+
+        if len(pt_lat1) != len(pt_hgt):
+            raise AeroAnsError(
+                'Extract_pt:Mismatch in number of points for lat/long/height')
+
+        # Check that heights have been merged with orography.
+        if not icube.coords('altitude'):
+            raise AeroAnsError(
+                'Extract_pt:Height requested but input data does not contain \
+                "Altitude" coordinate')
+        else:
+
+            # Store the min and max altitudes from cube data so that user
+            # cannot request points located below/ above that.
+            # Will extract =min/max if beyond limits.
+            hgt_min = icube.coord('altitude').points.min()
+            hgt_max = icube.coord('altitude').points.max()
+
+    # ---------- Finished checks -- begin processing -------------------------
+
+    # If level specified, extract slice first
+    if 'level' in keys:
+
+        if icube.coord('model_level_number'):
+            icube = icube.extract(
+                iris.Constraint(model_level_number=keys['level']))
+
+        else:
+
+            if icube.coord('pseudo_level'):
+
+                icube = icube.extract(
+                    iris.Constraint(pseudo_level=keys['level']))
+
+    # Extract values for specified points lat/lon
+
+    # ******** Does not seem to handle multiple points if 3-D *************
+
+    data_out = []
+
+    for n in np.arange(len(pt_lat1)):
+        pt_coords = [('latitude', pt_lat1[n]), ('longitude', pt_lon1[n])]
+
+        if 'nearest' in keys and keys['nearest']:
+            tcube = icube.interpolate(pt_coords, iris.analysis.Nearest())
+
+        else:
+            tcube = icube.interpolate(pt_coords, iris.analysis.Linear())
+
+        # Extract at requested height
+
+        if 'height' in keys:
+            pt = max(pt_hgt[n], hgt_min)
+            pt = min(pt_hgt[n], hgt_max)
+            pt_coords = [('altitude', pt)]
+
+            if 'nearest' in keys and keys['nearest']:
```

Comment:
note that you are still inside the loop on `n` but looking at the coordinate you want to interpolate on, I see it's only altitude, do you not want to move this block outside the `n` loop? It looks to me the interpolation operation is repeated n times for the same result

Reply:
The interpolation should indeed be out of the loop. Thank you for picking that one up!